### PR TITLE
Upgrade to Picard 2.23.3

### PR DESCRIPTION
**Pull request by cmnbroad**

### CODE REVIEW

**Comment by cmnbroad** [comment]
Yes, it does. Would you call that a bug or a feature ?

File: src/main/java/org/broadinstitute/hellbender/Main.java [hunk 1]
```diff
@@ -43,10 +43,6 @@
          * to think about number formatting issues.
          */
         Utils.forceJVMLocaleToUSEnglish();
-
-        // Turn off the Picard legacy parser and opt in to Barclay syntax for Picard tools. This should be replaced
-        // with a config setting once PR https://github.com/broadinstitute/gatk/pull/3447 is merged.
-        System.setProperty("picard.useLegacyParser", "false");
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/Main.java:49]: Do we actually want to allow users to run Picard tools via GATK using the legacy syntax @cmnbroad ?

**Comment by lbergelson** [comment]
I'd call it a bug since we would like to remove the legacy parser at some point but I could be convinced either way.

**Comment by cmnbroad** [comment]
I was hoping to get rid of the remaining code in Picard that supports this, but I've reverted it for now. Back to you @droazen .

**Pull request merged**